some features for the paper

o- opened the pull request:
1. Add a disassembler to produce ott compatible output that can be c/p into the tex
2. Untangle some optimizations to be able to show the changes in smaller increments.

o- committed:
add ott compatible disassembly output

as an example fix variable names in inline_osr to work with latex.

o- committed:
untangle some optimizations

mhyee commented on transform.ml:
```diff
@@ -130,22 +139,17 @@ let optimize (opts : string list) (prog : program) : program option =
         optimizer;
         Transform_assumption.remove_empty_osr;
-        optimizer_classic;
+        optimizer;
```
Remind me again, why do we need `optimizer` repeatedly here?

o- replied:
no good answer...

some transformations might enable other, so running multiple times makes more optimizations.

maybe we could instead have a cli argument for it...

mhyee reviewed (commented):
It looks like you forgot to add some files (`ott.ml`?).

o- committed:
make number of optimizer runs configurable

o- commented:
ok, now I added configurable optimizer pass number.
Also added the missing ott.ml file.
The code duplication is intentional since for now it would have been a bigger refactoring to make the disassembler configurable.

o- commented:
the ott syntax is a bit strange because I need to differ between normal instruction rendering and aligned instruction rendering.

The normal syntax of a program is:
```
fun1 -> (x) ( version1 -> ( l1 -> return x ) ) ),
fun2 -> (y) ( versionx -> ( l1 -> return y ) ) )
```

If you want aligned instructions (ie. the instruction rendered as `l1 & : & return x` you can enclose the program in `P( ... )`, or a plain list of labeled instructions in `i( ... )`.

mhyee commented on ott.ml:
```diff
@@ -0,0 +1,153 @@
+        in
+        let dump_frame buf {cont_pos={func; version; pos}; cont_res; varmap} =
+          pr buf "(%s, %s, %s) [%s = $, %a]"
+              func version pos
```
Nitpick: indentation should be 2 spaces not 4. (See the `pr buf ...` below.)

mhyee commented on ott.ml:
```diff
@@ -0,0 +1,153 @@
+      let print_formals buf = (dump_comma_separated print_formal) buf formals in
+      Printf.bprintf buf "%s -> ( %t ) ( " name print_formals;
+      let dump_vers buf version =
+          pr buf "%s -> (\n" version.label;
```
Nitpick: should be 2 space indentation.

mhyee commented on sourir.ml:
```diff
@@ -13,6 +15,8 @@ let () =
     ("--quiet", Arg.Set quiet, "quiet");
     ("--autofix", Arg.Set autofix, "automatically normalize graph");
     ("--opt", Arg.String (fun s -> opts := String.split_on_char ',' s), "Enable optimizations");
+    ("--nopt", Arg.String (fun s -> opt_runs := int_of_string s), "Number of optimization runs");
```
Could you rename this to `opt-runs` or `num-opts`? `nopt` looks too much like `nop` or `no opt` to me.

mhyee commented on ott.ml:
```diff
@@ -0,0 +1,153 @@
+
+let disassemble buf (prog : Instr.program) =
+  let dump_func buf {name; formals; body} =
+      let print_formal buf (Param x) = pr buf "%s " x in
```
Nitpick: should be 2 space indentation.

o- committed:
fix ott output
